### PR TITLE
Adding support to link listen ports with private keys

### LINKED ISSUE #166: Support for Private Keys on Listen Ports
Add support for Private Keys on Listen Ports.
By the time Listen ports were supported, private keys were not available yet.

### PR DESCRIPTION
**Pull request by joao-borges**
Fixes #166 

## Proposed Changes
* Adding `privateKey` to listen-ports configuration file, which has to point to an existing private key available in the bundle.

### CODE REVIEW

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/ListenPortEntityBuilder.java [hunk 1]
```diff
@@ -101,6 +106,14 @@ Entity buildListenPortEntity(Bundle bundle, String name, ListenPort listenPort,
             Element tlsSettingsElement = document.createElement(TLS_SETTINGS);
             tlsSettingsElement.appendChild(createElementWithTextContent(document, CLIENT_AUTHENTICATION, tlsSettings.getClientAuthentication().getType()));
 
+            if (isNotEmpty(tlsSettings.getPrivateKey())) {
+                final PrivateKey privateKey = bundle.getPrivateKeys().get(tlsSettings.getPrivateKey());
```
Comment (J-Lou) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/ListenPortEntityBuilder.java]: How are these private keys set?

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/loader/ListenPortLoader.java [hunk 2]
```diff
@@ -35,7 +39,13 @@ public void load(final Bundle bundle, final Element element) {
         final Set<String> enabledFeatures = getChildElementsTextContents(getSingleChildElement(listenPortElement, ENABLED_FEATURES), STRING_VALUE);
         final String targetServiceReference = getSingleChildElementTextContent(listenPortElement, TARGET_SERVICE_REFERENCE);
         final ListenPortTlsSettings tlsSettings = getTlsSettings(listenPortElement);
-        final Map<String, Object> properties = mapPropertiesElements(getSingleChildElement(listenPortElement, PROPERTIES, true), PROPERTIES);
+        Map<String, Object> properties = mapPropertiesElements(getSingleChildElement(listenPortElement, PROPERTIES, true), PROPERTIES);
+        if (!equalsAny(protocol, PROTOCOL_FTP, PROTOCOL_FTPS)) {
```
Comment (J-Lou) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/loader/ListenPortLoader.java]: What are the intended values for protocol?

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/ListenPortEntityBuilder.java [hunk 3]
```diff
@@ -42,7 +47,7 @@
 
     private static final Map<String, ListenPort> DEFAULT_PORTS = unmodifiableMap(createDefaultListenPorts());
     private static final String USES_TLS = "usesTLS";
-    private static final Integer ORDER = 800;
+    private static final Integer ORDER = 1200;
```
Comment (J-Lou) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/ListenPortEntityBuilder.java]: What is integer change for?

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/ListenPortEntityBuilder.java (hunk 1 above)
Comment (joao-borges) [reply config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/ListenPortEntityBuilder.java]: private key metadata is in json/yaml similar to all environment entities

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/loader/ListenPortLoader.java (hunk 2 above)
Comment (joao-borges) [reply config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/loader/ListenPortLoader.java]: can be HTTP, HTTPS, SSH, FTP or FTPS
but in this case I am removing property that does not make sense if not FTP(s)

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/ListenPortEntityBuilder.java (hunk 3 above)
Comment (joao-borges) [reply config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/ListenPortEntityBuilder.java]: ListenPort builder has to be invoked after the private key builder, so it finds the key into the bundle.

**Pull request merged**